src/ElmFormat/Filesystem.hs

```diff
@@ -63,3 +64,7 @@ findAllElmFiles inputFile =
 hasFilename :: String -> FilePath -> Bool
 hasFilename name path =
     name == FilePath.takeFileName path
+
+addElmVersion :: FileStore f => FilePath -> Free f (FilePath, ElmVersion)
```

Comment:
You might prefer other names for this function and `ElmVersion.fromFile`, I'm open to suggestions if you don't like them.

Reply:
Yeah, the name seems good.  The one thing I wonder about is how it might work out to have `findAllElmFiles` return `[ElmFile]` instead of `[FilePath]` and having to use `addElmVersion` later.

Reply:
Also having `findAllElmFiles` return `[ElmFile]` would allow a future optimization of detecting the version at the top and know it already as you recurse down into directories instead of later having to recurse up from each file to find its version.

Reply:
I added the optimization in the last version.